### PR TITLE
2.0 support + no-game export

### LINKED ISSUE #9: Custom tech is detected as pickup nodes in rdv
Since these tech are added as regular, no pre-reqs tech that costs only 10 red science, the automated db import creates a pickup node for these.

### LINKED ISSUE #11: Include all custom assets instead of generating in runtime
This lets exporting without having a copy of the game, improving workflow.
Ideally, all assets should be packed in a "randovania-assets" mod, which is downloaded in runtime.

### PR DESCRIPTION
**Pull request by henriquegemignani**
Now uses the 2.0 API, dropping support to 1.1 and older.
All custom assets used by the randomizer now comes from a dedicated [assets mod](https://github.com/randovania/factorio-assets-mod) that is downloaded from the mod portal. 
Now with 2.0, the game detects missing dependencies and will prompt you to fix it. Testing it, it does takes some unusual steps so maybe a good thing to download it ourselves if it's missing.

With the assets mod and smart use of the modding APIs, it's no longer necessary to fetch any data from the game when creating the randomizer mod.

Fixes #9.
Fixes #11.

### CODE REVIEW

**Comment by Miepee** [comment]
is it intended that the tests fail?

**Comment by henriquegemignani** [comment]
Apparently I forgot to delete a test file for a removed file 🤣

File: src/factorio_randovania_mod/lua_src/prototypes/tech.lua [hunk 1]
```diff
@@ -6,22 +9,28 @@ local function make_tech(param)
...
     end
+    param.randovania_custom_tech = true
+    assert(param.unit)
     return param
 end
 
+local impossible_lab = table.deepcopy(data.raw["lab"]["lab"])
+impossible_lab.name = "impossible-lab"
+impossible_lab.inputs = { "impossible-science-pack" }
+
```
Comment (duncathan) [inline src/factorio_randovania_mod/lua_src/prototypes/tech.lua:20]: add a comment explaining why this is necessary since it isn't exactly obvious imo

**Review by duncathan — CHANGES_REQUESTED** [review]
I feel like I'm just missing it, but what unlocks pipes now?

**Comment by henriquegemignani** [comment]
> I feel like I'm just missing it, but what unlocks pipes now?

If I didn't mess anything, you start with them unlocked.

**Review by duncathan — APPROVED** [review]

**Pull request merged**